peer_test.go

```diff
@@ -539,3 +550,30 @@ func validateStressTest(t testing.TB, server *Channel, numAffinity int) {
 	median := counts[len(counts)/2]
 	testDistribution(t, countsByPeer, float64(median)*0.9, float64(median)*1.1)
 }
+
+func (pt *peerSelectionTest) runAfterConnClosed() {
```

Comment:
I don't think this should be part of this test.

Make a separate test that tests only this behaviour. That test doesn't need to be part of the stress test, and should run along with all other tests.